### PR TITLE
Merge validate journal code

### PR DESCRIPTION
**Pull request by ryscher**
An attempt at resolving the merge conflicts for #999. 

(And also cleanup of handling for the Context objects)

### CODE REVIEW

**Comment by daisieh** [comment]
Yep, I just looked at it and it looks fine to me.

File: dspace-api/src/main/java/org/dspace/content/authority/Concept.java [hunk 1]
```diff
@@ -445,10 +446,16 @@ public static Concept find(Context context, int id) throws SQLException
...
     {
         ArrayList<Concept> concepts = new ArrayList<Concept>();
-        TableRowIterator row = DatabaseManager.query(context,"select c.* from concept as c, conceptmetadatavalue as cmv where cmv.text_value = '"+journalID+"' and cmv.parent_id=c.id;");
+        String schema = "journal";
+        String element = "journalID";
+        MetadataSchema mds = MetadataSchema.find(context, schema);
+        MetadataField mdf = MetadataField.findByElement(context, mds.getSchemaID(), element, null);
+        journal_field_id = mdf.getFieldID();
+        log.info ("journal field id is " + journal_field_id);
+        TableRowIterator row = DatabaseManager.query(context, "select c.* from concept as c, conceptmetadatavalue as cmv where upper(cmv.text_value) = ? and cmv.parent_id = c.id and cmv.field_id = ?;", journalID, journal_field_id);
```
Comment (ryscher) [inline dspace-api/src/main/java/org/dspace/content/authority/Concept.java]: This assumes that journalID is uppercase, but there isn't code in this method to ensure it is uppercase.

Comment (daisieh) [inline dspace-api/src/main/java/org/dspace/content/authority/Concept.java]: Can you fix this directly in your branch?

**Comment by ryscher** [comment]
This looks good now. Merging.

**Pull request merged**